test/test_complex.py

```diff
@@ -1,13 +1,35 @@
+import math
 import torch
 from torch.testing._internal.common_utils import TestCase, run_tests
+import unittest
+
+TEST_NUMPY = True
+try:
+    import numpy as np
+except ImportError:
+    TEST_NUMPY = False
 
 devices = (torch.device('cpu'), torch.device('cuda:0'))
 
+
 class TestComplexTensor(TestCase):
     def test_to_list_with_complex_64(self):
         # test that the complex float tensor has expected values and
         # there's no garbage value in the resultant list
         self.assertEqual(torch.zeros((2, 2), dtype=torch.complex64).tolist(), [[0j, 0j], [0j, 0j]])
 
+    @unittest.skipIf(not TEST_NUMPY, "Numpy not found")
+    def test_exp(self):
+        def exp_fn(dtype):
+            a = (1j * torch.tensor([0, 1 + 1j, 2 + 2j, 3 + 3j], dtype=dtype) /
```

Comment:
this input is too small to trigger vectorized kernels. Also, please put the test into test_torch.py, TestTorchMathOps now supports testing operations against numpy reference implementation https://github.com/pytorch/pytorch/pull/35658/files. You'd need to modify _generate_reference_input to generate complex inputs for corresponding dtypes.

Reply:
oh good to know @ngimel 

@aayn let's keep this test here (and also link the issue in a comment above) and add another test in test_torch.py as @ngimel suggested

Reply:
@ngimel @anjali411 

I added testing for complex exp as @ngimel suggested. However, `torch.isclose` is not implemented for complex types. So [this line](https://github.com/pytorch/pytorch/blob/master/torch/testing/__init__.py#L30) throws an error `RuntimeError: "le_cpu" not implemented for 'ComplexFloat'`. All other testcases pass because they use `assertEqual` from `TestCase`.

Reply:
hm, makes sense to implement `.isclose` for complex, but that's out of scope for this PR.

Reply:
@aayn 

Yes, `<` is not implemented in `std::complex<T>` or `PyComplex`. I converted to my tenors to numpy arrays and used numpy tests `assert_allclose` to test. Take a look [here](https://gitlab.com/pytorch-complex/pytorch-cpu-strided-complex/-/blob/master/test/test_torch.py). My unit tests were very crude.

Also, you need to have test vectors with (256/16 bit)  + 1 = 17 elements to trigger vectorized and scalar functions for half, float, and double precision. We don't actually support `std::complex<half>` yet though.